notebooks/ch12/linear_projection.md

```diff
@@ -0,0 +1,330 @@
+
+Visually,
+
+<img src="../../notebooks-images/linear_projection1.png" width="500" />
+
+Since our error is based on distance, we can see that to minimize $ L(\hat{\theta}, \vec{x}, \vec{y}) = \left \Vert  \vec{y} - X \hat{\theta} \right \Vert ^2$, we want $X \hat{\theta}$ to be as close to $\vec{y}$ as possible.
+
+Mathematically, we are looking for the projection of $\vec{y}$ onto the vector space spanned by the columns of $X$ because the projection $\hat{y}$ has the property of being the closest point in the vector space to the vector $y$.
+<img src="../../notebooks-images/linear_projection2.png" width="500" />
+
+To see why, consider other points on the vector space, in purple.
+<img src="../../notebooks-images/linear_projection3.png" width="500" />
+
+By the Pythagorean Theorem, any other point on the plane is farther from $\vec{y}$ than $\hat{y}$ is. The longer the distance, the higher the least squared error. Thus, the length of the perpendicular represents the least squared error.
+
+## Linear Algebra
+
+Since we've snuck in a lot of linear algebra concepts already, all that's left is solving for the $\hat{\theta}$ that yields our desired $\hat{y}$.
+
+A couple things to note: 
+
+<img src="../../notebooks-images/linear_projection5.png" width="500" />
+
+- $\hat{y} + e = \vec{y}$
+- $\vec{e}$ is perpendicular to $\vec{x}$ and $\vec{1}$
+- $\hat{y} = X \hat{\theta^*}$ is the vector closest to $y$ in the vector space spanned by $\vec{x}$ and $\vec{1}$
+
+Thus, we arrive at the equation:
+
+$$X  \hat{\theta^*} + \vec{e} = \vec{y}$$
+
+Left-multiplying both sides by $X^T$:
+
+$$X^T X  \hat{\theta^*} + X^T \vec{e} = X^T \vec{y}$$
+
+But since $\vec{e}$ is perpendicular to the columns of $X$, $X^T \vec{e}$ is a column vector of $0$'s. Thus, we arrive at the Normal Equation:
+
+$$X^T X  \hat{\theta^*} = X^T \vec{y}$$
+
+From here, we can easily solve for $\hat{\theta^*}$ by left-multiplying both sides by $(X^T X)^{-1}$:
+
+$$\hat{\theta^*} = (X^T X)^{-1} X^T \vec{y}$$
+
+## Finishing up the Case Study
+
+Let's return to our case study and apply what we've learned, and explain to our client why our solution is sound.
+
+$$
+\vec{y} = \begin{bmatrix} 2 \\ 1 \\ -2  \end{bmatrix} \qquad X = \begin{bmatrix} 1 & 3 \\ 1 & 0 \\ 1 & -1 \end{bmatrix}
+$$
+
+
+$$
+\begin{align}
+\hat{\theta^*} 
+&= 
+\left(
+\begin{bmatrix} 1 & 1 & 1 \\ 3 & 0 & -1 \end{bmatrix}
+\begin{bmatrix} 1 & 3 \\ 1 & 0 \\ 1 & -1 \end{bmatrix}
+\right)^{-1}
+\begin{bmatrix} 1 & 1 & 1 \\ 3 & 0 & -1 \end{bmatrix}
+\begin{bmatrix} 2 \\ 1 \\ -2  \end{bmatrix} \\
+&= 
+\left(
+\begin{bmatrix} 3 & 2\\ 2 & 10 \end{bmatrix}
+\right)^{-1}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{30-4}
+\begin{bmatrix} 10 & -2\\ -2 & 3 \end{bmatrix}
+\begin{bmatrix} 1 \\ 8 \end{bmatrix} \\
+&=
+\frac{1}{26}
+\begin{bmatrix} -6 \\ 22 \end{bmatrix}\\
+&=
+\begin{bmatrix} - \frac{3}{13} \\ \frac{11}{13} \end{bmatrix}
```

Comment:
![image](https://user-images.githubusercontent.com/22417757/41815162-5fc1e486-7717-11e8-8e5a-cbae2252b1dc.png)
This format still looks really wonky sad